linera-core/src/client.rs

```diff
@@ -190,6 +190,14 @@ impl<P, S> ChainClient<P, S> {
         }
     }
 
+    pub async fn with_notifier_and_cache(mut self, other: &Self) -> Self {
```

Comment:
Happy to see this removed in the next commit!

Reply:
(Yes, I just wanted to try out quickly whether that fixed the deadlock.)